pkg/spdx/implementation.go

```diff
@@ -232,19 +236,32 @@ func getImageReferences(referenceString string) ([]ImageReferenceInfo, error) {
 			return nil, fmt.Errorf("while calculating image digest: %w", err)
 		}
 
-		dig, err := name.NewDigest(
-			fmt.Sprintf(
-				"%s/%s@%s:%s",
-				tag.RegistryStr(), tag.RepositoryStr(),
-				imageDigest.Algorithm, imageDigest.Hex,
-			),
-		)
-		if err != nil {
-			return nil, fmt.Errorf("building single image digest: %w", err)
+		// If we could not turn the reference to digest then we synthesize one
+		if tok && !dok {
+			dig, err = name.NewDigest(
+				fmt.Sprintf(
+					"%s/%s@%s:%s",
+					tag.RegistryStr(), tag.RepositoryStr(),
+					imageDigest.Algorithm, imageDigest.Hex,
+				),
+			)
+			if err != nil {
+				return nil, fmt.Errorf("building single image digest: %w", err)
+			}
 		}
 
 		logrus.Infof("Adding image digest %s from reference", dig.String())
-		return append(images, ImageReferenceInfo{Digest: dig.String()}), nil
+		images.Digest = dig.String()
+		mt, err := im.MediaType()
+		if err == nil {
+			images.MediaType = string(mt)
+		}
+		conf, err := im.ConfigFile()
+		if err == nil {
+			images.Arch = conf.Architecture
+			images.OS = conf.OS
+		}
+		return images, nil
 	}
```

Comment:
Since we do `images.Digest = dig.String()` in the block above, I'd say we move the code in L219 down below this block.

Reply:
This logic was refactored into the new index function  :+1: